addons/website_sale_slides/models/slide_channel.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields
+
+
+class Channel(models.Model):
+    _inherit = 'slide.channel'
+
+    visibility = fields.Selection(selection_add=[('payment', 'On payment')])
+    product_id = fields.Many2one('product.product', 'Product')
```

Comment:
As we make a search on channels with product_id in domain I think we can add an index on it, even if we should not have tons of channels.